### PR TITLE
Write 1 in the ts-numbers when series.width == 1

### LINKED ISSUE #587: Antares V8.2-rc2 - Incorrect TS numbers in the ts-numbers output directory
**Description**
In the attached study containing 2 links, there is only one Time-Series for the link area-area 3, and the scenario builder NTC TS are set at random for the 4 MC years of the study.
However, when running a simulation, the ts-numbers output directory contains the following vector for the area - area 3 link: (4, 4, 2, 3).

**Expected behavior**
The ts-numbers output text files should contain only TS numbers corresponding to the input TS numbers for a given links (in this specific case, (1, 1, 1, 1)).

**Screenshots and additional files**
[Initial_study_for_datastorm.zip](https://github.com/AntaresSimulatorTeam/Antares_Simulator/files/8196045/Initial_study_for_datastorm.zip)

**OS and version**
 - OS: Windows
 - Antares Simulator version: 8.2-RC2

### PR DESCRIPTION
**Pull request by flomnes**
Before that patch, random numbers were drawn, ignored in the simulation, but still written in ts-numbers.

### CODE REVIEW

File: src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp [hunk 1]
```diff
@@ -65,33 +65,44 @@ static void InitializeTimeSeriesNumbers_And_ThermalClusterProductionCost(
 
         // Load
         {
-            const Data::DataSeriesLoad& data = *area.load.series;
+            Data::DataSeriesLoad& data = *area.load.series;
             assert(year < data.timeseriesNumbers.height);
-            ptchro.Consommation
-              = (data.series.width != 1) ? (long)data.timeseriesNumbers[0][year] : 0; // zero-based
+            if (data.series.width == 1)
```
Comment (guilpier-code) [inline src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp]: As **timeseriesNumbers** was set previously in **timeseries-numbers.cpp**, it should already be set to **0** if there is only one available time series for some given area and year.
Said in another way, there should be a need for setting **timeseriesNumbers** here.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp (hunk 1 above)
Comment (flomnes) [reply src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp]: What you mean is that `timeseriesNumbers` should be set to 0 if `width==1` in **timeseries-numbers.cpp**. Then we would introduce a dependency to timeseries in **timeseries-numbers.cpp**, I'm not sure we want to do that.

Comment (flomnes) [reply src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp]: Actually, that dependency already exists in **timeseries-numbers.cpp**. Therefore, the TS numbers provided by `TimeSeriesNumbers::Generate` are wrong when `width==1`. I agree : the fix should be applied there, not here.

**Comment by guilpier-code** [comment]
As you now set **timeseriesNumbers** to **0** when there is only **1** TS for given year, area, ... in **timeseries-numbers.cpp**, file **alea_tirage_au_sort_chroniques.cpp** can be simplified. 
Example : 
```c++
ptchro.Consommation = data.timeseriesNumbers[0][year];
```

File: src/tests/src/solver/simulation/tests-ts-numbers.cpp [hunk 2]
```diff
@@ -354,7 +353,6 @@ BOOST_AUTO_TEST_CASE(two_areas_3_renew_clusters_with_respectively_4_4_1_ready_ma
 	// TS number checks
 	uint year = 0;
 	BOOST_CHECK_EQUAL(rnCluster_12->series->timeseriesNumbers[0][year], rnCluster_11->series->timeseriesNumbers[0][year]);
-	BOOST_CHECK_EQUAL(rnCluster_21->series->timeseriesNumbers[0][year], rnCluster_11->series->timeseriesNumbers[0][year]);
```
Comment (guilpier-code) [inline src/tests/src/solver/simulation/tests-ts-numbers.cpp:357]: ok, so before the patch, **timeseriesNumbers** is to something != 0 even if the TS actual size is **1**.
The patch in this test makes **rnCluster_21** useless. And we should rename that test.
And maybe make another test that show te new behavior.

File: src/tests/src/solver/simulation/tests-ts-numbers.cpp [hunk 3]
```diff
@@ -233,7 +233,6 @@ BOOST_AUTO_TEST_CASE(two_areas_3_thermal_clusters_with_respectively_4_1_4_ready_
 
 	// TS number checks
 	uint year = 0;
-	BOOST_CHECK_EQUAL(thCluster_12->series->timeseriesNumbers[0][year], thCluster_11->series->timeseriesNumbers[0][year]);
```
Comment (guilpier-code) [inline src/tests/src/solver/simulation/tests-ts-numbers.cpp:236]: Same remark as below (for renewable clusters)

**Review by guilpier-code — APPROVED** [review]

File: src/tests/src/solver/simulation/tests-ts-numbers.cpp (hunk 3 above)
Comment (guilpier-code) [reply src/tests/src/solver/simulation/tests-ts-numbers.cpp:236]: bla

File: src/tests/src/solver/simulation/tests-ts-numbers.cpp (hunk 2 above)
Comment (flomnes) [reply src/tests/src/solver/simulation/tests-ts-numbers.cpp:357]: I will just remove these obsolete clusters and rename the corresponding tests. My fix does not change the behavior of `Antares::Solver::TimeSeriesNumbers::Generate`.

The correct way to fix the bug would be to change `Antares::Solver::TimeSeriesNumbers::Generate`, but unfortunately it is called before the timeseries generator (the one that generates the coefficients of the TS, not the numbers). A large refactor would be necessary to include that change in `TimeSeriesNumbers::Generate`.

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**